string.go

```diff
@@ -16,11 +22,30 @@ func (s String) Errorf(format string, a ...interface{}) error {
 	}
 }
 
+// Wrap preserves the existing error's string value. It just serves as
+// a way to annotate an error so that errors.Is will return true when
+// compared to the same base string
```

Comment:
```suggestion
// Wrap returns nil when passed nil, otherwise it returns an error such that [errors.Is] returns true when compared to the String error or the passed in error.  It leaves the error text unchanged.
```